lib/tracer/rooted.go

```diff
@@ -0,0 +1,29 @@
+package tracer
+
+import (
+	"context"
+	"encoding/binary"
+	"hash/fnv"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// StartChainHeight returns a context and span rooted to the chain+height.
+// This creates a new trace root and should generally only by xprovider or cprovider.
+func StartChainHeight(ctx context.Context, chain string, height uint64, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
+	// Deterministic TraceID for chain+height. So all traces across all instances are correlated.
```

Comment:
I thought that TraceID is a primary key. If we have multiple traces with the same ID, would the insert work 🤔

Reply:
the idea is to correlate traces for the same "job"